model-engine/model_engine_server/api/llms_v1.py

```diff
@@ -384,3 +388,41 @@ async def download_model_endpoint(
             status_code=404,
             detail="The requested fine-tuned model could not be found.",
         ) from exc
+
+
+@llm_router_v1.delete(
+    "/model-endpoints/{model_endpoint_name}", response_model=DeleteLLMEndpointResponse
+)
+async def delete_llm_model_endpoint(
+    model_endpoint_name: str,
+    auth: User = Depends(verify_authentication),
+    external_interfaces: ExternalInterfaces = Depends(get_external_interfaces),
+) -> DeleteLLMEndpointResponse:
+    add_trace_resource_name("llm_model_endpoints_delete")
+    logger.info(f"DELETE /model-endpoints/{model_endpoint_name} for {auth}")
+    try:
+        use_case = DeleteLLMEndpointByNameUseCase(
+            llm_model_endpoint_service=external_interfaces.llm_model_endpoint_service,
+            model_endpoint_service=external_interfaces.model_endpoint_service,
+        )
+        return await use_case.execute(user=auth, model_endpoint_name=model_endpoint_name)
+    except (ObjectNotFoundException) as exc:
+        raise HTTPException(
+            status_code=404,
+            detail="The requested model endpoint could not be found.",
+        ) from exc
+    except (ObjectNotAuthorizedException) as exc:
+        raise HTTPException(
+            status_code=401,
+            detail="You don't have permission to delete the requested model endpoint.",
+        ) from exc
+    except ExistingEndpointOperationInProgressException as exc:
+        raise HTTPException(
+            status_code=409,
+            detail="Existing operation on endpoint in progress, try again later.",
+        ) from exc
+    except EndpointDeleteFailedException as exc:  # pragma: no cover
+        raise HTTPException(
+            status_code=500,
+            detail="deletion of endpoint failed, compute resources still exist.",
```

Comment:
Don't think we need to or should mention details about compute resources existing.